Added disabled state to autocomplete and tweaked lookup's disabled state visually

Blackbaud-TrevorBurch opened the pull request:
#99

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable)

Blackbaud-AlexKingman commented on e2e/autocomplete.e2e-spec.ts:
```diff
@@ -56,4 +56,19 @@ describe('Autocomplete', () => {
+    SkyHostBrowser.setWindowBreakpoint('xs');
+    element(by.css('#autocomplete-reactive-disable-button')).click();
+    expect('#autocomplete-reactive').toMatchBaselineScreenshot(done, {
+      screenshotName: 'autocomplete-disabled'
```
Should this be `autocomplete-disabled-xs` ?

Blackbaud-TrevorBurch replied:
Yup. Good call.

Blackbaud-AlexKingman commented on src/app/public/modules/autocomplete/autocomplete-input.directive.ts:
```diff
@@ -4,7 +4,8 @@ import {
   OnInit,
-  Renderer2
+  Renderer2,
+  Input
```
Check your alphabetical ordering here.

Blackbaud-TrevorBurch replied:
Fixed

Blackbaud-AlexKingman commented on src/app/visual/autocomplete/autocomplete-visual.component.html:
```diff
@@ -51,6 +51,14 @@ <h2>
 
+  <button
+    id="autocomplete-reactive-disable-button"
+    class="sky-btn sky-btn-primary"
```
Check your alphabetical ordering here.

Blackbaud-TrevorBurch replied:
Fixed. Also fixed a couple other places in this file that were wrong historically.

Blackbaud-AlexKingman commented on src/app/visual/autocomplete/autocomplete-visual.component.ts:
```diff
@@ -45,6 +49,20 @@ export class AutocompleteVisualComponent implements OnInit {
+    this.reactiveDisabledState = !this.reactiveDisabledState;
+  }
+
+  public toggleTemplateDisabled() {
```
Methods need a return type.

Blackbaud-TrevorBurch replied:
Fix

Blackbaud-TrevorBurch commented:
@Blackbaud-AlexKingman addressed your comments. 

As for your question, that behavior matches what our other inputs do today when disabled and I verified that non-Sky inputs also are marked invalid when they are disabled.

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable)

blackbaud-johnly commented on src/app/public/modules/autocomplete/autocomplete-input.directive.ts:
```diff
@@ -53,6 +54,20 @@ const SKY_AUTOCOMPLETE_VALIDATOR = {
 })
 export class SkyAutocompleteInputDirective implements OnInit, OnDestroy, ControlValueAccessor, Validator {
 
+  @Input()
```
Should we go ahead and add a JSDoc comment here? I assume it just needs to say: `Indicates whether to disable the autocomplete field.` The autocomplete docs aren't updated to use JSDocs yet, but I think it's a good practice to add JSDoc comments as we go. I created https://github.com/blackbaud/skyux2-docs/issues/953 to document this new property in the existing docs and will handle that, but I think we should also add the JSDoc here as part of the dev work. Let me know if my suggested description doesn't work.

blackbaud-johnly replied:
I set https://github.com/blackbaud/skyux2-docs/pull/954 to update the docs.

Blackbaud-TrevorBurch replied:
Done

blackbaud-johnly replied:
Looks good!

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable)

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable)
